src/devices/Common/Iot/Device/Common/GreatCircle.cs

```diff
@@ -0,0 +1,356 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using UnitsNet;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// This class provides some basic geodetic calculations on the WGS84 ellipsoid.
+    /// This returns exact distances between two points, considering the non-spherical shape of the earth.
+    /// The results are accurate for distances greater than about 1cm.
+    /// </summary>
+    public static class GreatCircle
```

Comment:
I'm not sure I like the name; I like the original name `GeoidCalculations` though - I'd recommend to put the other file in different namespace and re-use that name

Reply:
The problem is, that that would not be unique. We might also have distance calculations based on loxodromes instead of the great circle. But let's discuss that together with improved namings separatelly.